porcelain/protocol.go

```diff
@@ -23,7 +32,31 @@ func ProtocolParameters(plumbing protocolParamsPlumbing) (*ProtocolParams, error
 		return nil, errors.New("Failed to read autoSealInterval from config")
 	}
 
+	sectorSize, err := getSectorSize(plumbing)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not retrive sector size")
+	}
+
 	return &ProtocolParams{
 		AutoSealInterval: autoSealInterval,
+		SectorSizes:      []uint64{sectorSize},
 	}, nil
 }
+
+func getSectorSize(plumbing protocolParamsPlumbing) (uint64, error) {
+	var proofsMode types.ProofsMode
+	values, err := plumbing.MessageQuery(context.TODO(), address.Address{}, address.StorageMarketAddress, "getProofsMode")
```

Comment:
BLOCKING: `ProtocolParameters` needs to take a context parameter, this function needs to take a context, and it needs to be used in the call to MessageQuery.